crates/walrus-service/src/node.rs

```diff
@@ -1328,7 +1332,8 @@ impl StorageNode {
                 epoch_at_start,
                 first_new_event_epoch,
             );
-            self.inner.set_node_status(NodeStatus::RecoveryCatchUp)?;
+
+            self.enter_recovery_mode().await?;
```

Comment:
Fix 1: we need to cancel any ongoing blob sync even if this the first event we use to check node lagging. This is because there may exist prior blob extension events that skipped node lag checking and started blob sync.